.github/scripts/js/ci.js

```diff
@@ -79,116 +79,269 @@ module.exports.updateCommentOnStart = async ({ github, context, core, name }) =>
-
-    finalStatus = 'cancelled';
+  // Add a final workflow status and details about all jobs from the needs context.
+  if (statusConfig.includes('workflow')) {
+    let status = 'cancelled';
     let successCount = 0;
     let failureCount = 0;
-    for (const jobName in needsContext) {
-      if (!needsContext.hasOwnProperty(jobName)) {
+    for (const jobID in needsContext) {
+      if (!needsContext.hasOwnProperty(jobID)) {
+        continue;
+      }
+      // Ignore helper jobs.
+      if (jobID === 'started_at' || jobID === 'git_info') {
         continue;
       }
-      if (needsContext[jobName].result === 'success') {
+
+      let jobName = jobID;
+      if (jobNames && jobNames[jobID]) {
+        jobName = jobNames[jobID];
+      }
+      const jobResult = needsContext[jobID].result;
+
+      if (jobResult === 'success') {
         successCount++;
       }
-      if (needsContext[jobName].result === 'failure') {
+      if (jobResult === 'failure') {
         failureCount++;
       }
-      // Info about not started jobs.
-      if (needsContext[jobName].result === 'cancelled') {
-        jobsComment += `:ballot_box_with_check:\u00a0${jobName} cancelled.\n`;
+
+      // Info for not started job.
+      if ((jobResult === 'cancelled' || jobResult === 'skipped') && !statusConfig.includes(',no-skipped')) {
+        nonReportedJobs += renderJobStatusOneLine({ status: jobResult, name: jobName }) + `\n`;
       }
-      if (needsContext[jobName].result === 'skipped') {
-        jobsComment += `:ballot_box_with_check:\u00a0${jobName} skipped.\n`;
+
+      // Restore information for overridden job. Only result, no elapsed time here.
+      if ((jobResult === 'success' || jobResult === 'failure') && !comment.includes(`<!-- result-for: ${jobName} -->`)) {
+        let jobReport = '';
+        if (statusConfig.includes(',restore-one-line')) {
+          jobReport = renderJobStatusOneLine(jobResult, jobName);
+        } else if (statusConfig.includes(',restore-separate')) {
+          jobReport = renderJobStatusSeparate(jobResult, jobName);
+        }
+        nonReportedJobs += jobReport + `\n`;
       }
     }
     if (successCount > 0) {
-      finalStatus = 'success';
+      status = 'success';
     }
     if (failureCount > 0) {
-      finalStatus = 'failure';
-    }
-    if (jobsComment !== '') {
-      jobsComment = '\n' + jobsComment;
+      status = 'failure';
+      failedInfo = `${failureCount} job${failureCount > 1 ? 's' : ''} failed`;
+      core.setFailed(`Workflow ${name} failed: ${failedInfo}.`);
+      failedInfo = ` (${failedInfo})`;
     }
+
+    core.info(`Status for workflow report is ${status}`);
+
+    statusReport = renderWorkflowStatusFinal(status, name, ref, build_url, time_elapsed);
   }
 
-  console.log(`Status is ${finalStatus}`);
+  if (statusConfig.includes(',final')) {
+    // Cleanup first line with the response from the bot.
+    comment = comment.replace(/^Aye,.*\n/m, '');
```

Comment:
How can we reuse "Aye"? I'm afraid we can break something if we just change the comment text elsewhere.

Reply:
We can simply extract a constant with this regex and a template function for the comment. If we put them in one place, we can easily change them later since the connection will be obvious.